prov/util/src/util_wait.c

```diff
@@ -212,13 +275,20 @@ static int util_wait_fd_control(struct fid *fid, int command, void *arg)
 static int util_wait_fd_close(struct fid *fid)
 {
 	struct util_wait_fd *wait;
+	struct ofi_wait_fd_entry *fd_entry;
 	int ret;
 
 	wait = container_of(fid, struct util_wait_fd, util_wait.wait_fid.fid);
 	ret = fi_wait_cleanup(&wait->util_wait);
 	if (ret)
 		return ret;
 
+	while (!dlist_empty(&wait->fd_list)) {
+		dlist_pop_front(&wait->fd_list, struct ofi_wait_fd_entry,
+				fd_entry, entry);
+		free(fd_entry);
+	}
```

Comment:
Ideally, when the endpoint is closed, it should remove the fd entry from fd_list. This code takes care of the case when an endpoint doesn't do it. Perhaps this code can be removed as it's an error if the endpoint doesn't delete the fd from the list. May be we should use a reference counter on the util_wait object?